Spigot-API-Patches/0057-Add-PlayerUnknownCommandEvent.patch

```diff
@@ -0,0 +1,80 @@
+From 7ec63d75792b94c9de3593fe919799bb8c5e3d6e Mon Sep 17 00:00:00 2001
+From: Sweepyoface <github@sweepy.pw>
+Date: Wed, 14 Jun 2017 04:10:23 +0000
+Subject: [PATCH] Add PlayerUnknownCommandEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerUnknownCommandEvent.java b/src/main/java/org/bukkit/event/player/PlayerUnknownCommandEvent.java
+new file mode 100644
+index 0000000..2e78d5f
+--- /dev/null
++++ b/src/main/java/org/bukkit/event/player/PlayerUnknownCommandEvent.java
+@@ -0,0 +1,65 @@
++package org.bukkit.event.player;
++
++import org.bukkit.command.CommandSender;
++import org.bukkit.entity.Player;
++import org.bukkit.event.HandlerList;
++
++/**
++ * Thrown when a player executes a command that is not defined
++ */
++public class PlayerUnknownCommandEvent extends PlayerEvent {
++    private static final HandlerList handlers = new HandlerList();
++    private CommandSender sender;
++    private String commandLine;
++    private String message;
++
++    public PlayerUnknownCommandEvent(final CommandSender sender, final String commandLine, final String message) {
```

Comment:
no. This should accept `Player` if it is going to be a `Player`-only command. Accepting `CommandSender` is just stupid.